system_metrics_collector/src/system_metrics_collector/linux_process_memory_measurement_node.cpp

```diff
@@ -92,7 +94,7 @@ std::string LinuxProcessMemoryMeasurementNode::GetMetricName() const
 
 const std::string & LinuxProcessMemoryMeasurementNode::GetMetricUnit() const
 {
-  static const std::string unit_name{"percent"};
+  static const std::string unit_name{collector_node_constants::kPercentUnitName};
   return unit_name;
```

Comment:
I think its going to be better to return an implicitly converted `std::string` copy of `kPercentUnitName` by value. This creates storage for `unit_name` that has a lifecycle tied to the instance of `LinuxProcessMemoryMeasurementNode`. Anything that calls `LinuxProcessMemoryMeasurementNode::GetMetricUnit()` must ensure that the returned value does not outlive the instance of `LinuxProcessMeasurementNode` otherwise there is a memory error.

The compiler will probably optimiz

Reply:
Are you saying we should be lifecycling/churning `std::string` objects for the same constant value (`"percent"`)?

> This creates storage for `unit_name` that has a lifecycle tied to the instance of `LinuxProcessMemoryMeasurementNode`.

This is not true as `unit_name` is declared `static`.

Reply:
I got mixed up here, this is fine.